packages/template-retail-react-app/app/hooks/use-auth-modal.js

```diff
@@ -70,6 +74,20 @@ export const AuthModal = ({
     const login = useAuthHelper(AuthHelpers.LoginRegisteredUserB2C)
     const register = useAuthHelper(AuthHelpers.Register)
 
+    /************** merge basket for recurring users ***/
+    const prevAuthType = useRef()
```

Comment:
Would this be a bit cleaner?

```js
const usePrevious = (value) => {
  const ref = useRef()

  useEffect(() => {
    ref.current = value
  }, [value])

  return ref.current
}

const prevCustomerType = usePrevious(customerType)
```

Reply:
This is a very common pattern of using ref in React, is it worth creating a usePrevious hook? This means a separate file to maintain. By using directly in the files, we have a bit of duplication in 2 places, but that are not a lot

Reply:
Minor: I think it's worth creating a `usePrevious` hook for more readable code. 

To me, how the whole thing works is confusing. So having it abstracted away in a custom hook would be better for me.

And I think currently `prevAuthType.current === 'guest'` is not immediately clear because when I read the 'prev' and then 'current' in the same line.